lib/absinthe/phase/document/arguments/variable_types_match.ex

```diff
@@ -73,23 +130,20 @@ defmodule Absinthe.Phase.Document.Arguments.VariableTypesMatch do
 
     case Map.get(variable_defs, variable.name) do
       %{schema_node: variable_type} = variable_definition ->
-        if types_compatible?(
+        if !types_compatible?(
```

Comment:
I can clean this up if you want. This will work, but if we want a smaller diff we can revert this logic and _only_ include the Logger.warn line.

I think I'm going to do this quick before we merge this.